presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -142,14 +142,14 @@ public void destroy()
     }
 
     @Override
-    public List<SchemaTableName> getTableNames(@Nullable String schema)
+    public List<SchemaTableName> getTableNames(Optional<String> schema)
     {
         try (Connection connection = connectionFactory.openConnection()) {
             DatabaseMetaData metadata = connection.getMetaData();
-            if (metadata.storesUpperCaseIdentifiers() && (schema != null)) {
-                schema = schema.toUpperCase(ENGLISH);
+            if (metadata.storesUpperCaseIdentifiers()) {
+                schema = schema.map(schemaName -> schemaName.toUpperCase(ENGLISH));
             }
-            try (ResultSet resultSet = getTables(connection, schema, null)) {
+            try (ResultSet resultSet = getTables(connection, schema.orElse(null), null)) {
```

Comment:
`Accept Optional in BaseJdbcClient#getTableNames` commit should update `io.prestosql.plugin.jdbc.BaseJdbcClient#getTables` as well.

This would make `Annotate nullable paramters of BaseJdbcClient#getTables` commit unnecessary

Reply:
Then I would need to accept `Optional` in `escapeNamePattern` and handling that around `java.sql.DatabaseMetaData#getTables` that would make code to look cumbersome.

It is good to use `Optional` over `Nullable`, but let's do not go crazy with that.

Reply:
What's crazy about that?

Reply:
I think we should be consistent too

Reply:
For `DatabaseMetaData#getTables` it should just be an extra `.orElse(null)` at the end. I agree we should update everything, but I'm fine if we don't do it all at once, as each step is an improvement.

(but I also don't think doing everything in one shot is that much work)

Reply:
Fixed